de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/ui/views/buttons/ConnectButton.java

```diff
@@ -123,54 +142,202 @@ private void createAddAccountMenuItem() {
             @Override
             public void actionPerformed(ActionEvent e) {
                 XMPPAccount account = createNewAccount();
+
                 if (account == null) {
-                    SafeDialogUtils
-                        .showError("Account was not created.", "Error");
+                    return;
                 }
+
                 createMenuItems();
+
+                askToConnectToAccount(account);
             }
         });
     }
 
+    private void askToConnectToAccount(@NotNull XMPPAccount account){
+        try {
+            Integer option = SafeDialogUtils.showYesNoDialog(
+                Messages.ConnectButton_connect_to_new_account_message,
+                Messages.ConnectButton_connect_to_new_account_title);
+
+            if (option == com.intellij.openapi.ui.Messages.YES){
+                connectAction.executeWithUser(
+                    account.getUsername() + USERID_SEPARATOR
+                        + account.getDomain());
+            }
+
+
+        } catch (IllegalAWTContextException e) {
+            LOG.error("Account creation failed.", e);
+
+            SafeDialogUtils.showError(
+                "There was an error creating the account.\nDetails: "
+                    + e.getMessage(),
+                "Account creation failed");
+        }
+    }
+
     /**
-     * Asks for Name, Password and domain and server for a new XMPP account.
+     * Asks for user id (name@domain), password, and server for a new XMPP
+     * account.
+     *
+     * @return an <code>XMPPAccount</code> with the values entered by the user
+     *         or <code>null</code> if the creation was canceled by the user or
+     *         the user entered at least one illegal value
      */
-    protected XMPPAccount createNewAccount() {
-        final String userID = SafeDialogUtils.showInputDialog(
-            "Your User-ID, e.g. user@saros-con.imp.fu-berlin.de", "", "Login");
-        if (userID.isEmpty()) {
+    private XMPPAccount createNewAccount() {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. ([squid:S3776](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS3776))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=240&issues=AWYbBs1KvOWp7Yd63TpP&open=AWYbBs1KvOWp7Yd63TpP)

[ISSUE_KEY]: AWYbBs1KvOWp7Yd63TpP